14-5.3

chirayudesai opened the pull request.

grote committed:
Move MemoryLogger to core so we can use it in storage lib as well

grote committed:
Implement new Backend.save() method that can be re-tried

grote committed:
Use new Backend.save() method with app backup

grote committed:
Use new Backend.save() method with file backup

grote committed:
Remove old Backend.save() method and hacky WebDAV pipe code with it

chirayudesai committed:
Bump version to 15-5.2

Change-Id: I07a32331e2a310eb6f2b1dc6c65717ef6231c891

grote committed:
Implement new RetryBackend and wrap it around other backends

grote committed:
Add logging around StaleDataException

grote committed:
Update libs for Android 15 QPR1

grote committed:
File BackupService: don't restart automatically and log self stopping

grote committed:
Merge pull request #832 from grote/android15-qpr1

Update libs for gradle build for Android 15 QPR1

grote committed:
Introduce a BackendId, so RetryBackend doesn't mess up class comparisons

grote committed:
Allow BlobCreator's saving to be re-tried

grote committed:
Add test that snapshot saving can be re-tried

grote committed:
Ensure that file backup Snapshot saving can be re-tried

grote committed:
Ensure ChunkWriter saving can be re-tried

grote committed:
Don't schedule/cancel workers two times after setting new backend

grote committed:
Throw exception from SafHelper#findFileBlocking so RetryBackend can work with it

grote committed:
Add TODO to StorageBackup#init

grote committed:
Use UsbMonitorService in foreground to wait for USB root

See: https://developer.android.com/develop/background-work/background-tasks/broadcasts#android-14
for why we don't get broadcasts while the system has us in cached state

grote committed:
Fix: Convert app size from snapshot to old metadata

We forgot the app size, so when restoring apps, their data size was unknown

grote committed:
log logcat command at the beginning

grote committed:
log when file backup gets scheduled and add TODOs

grote committed:
Introduce SafRetryException to make SafBackend more resilient

grote committed:
Save build reports on cirrus CI

grote committed:
Remove keepTrying() for saving snapshots as we now have RetryBackend

grote committed:
Use linear delay for RetryBackend to make it less likely to run into system timeout

grote committed:
Treat UnknownHostException as transient, so it gets retried

grote committed:
Tell system to reinitialize when it refuses to give us data we need

Another hacky workaround to fix a hacky workaround.

We had talked to Google about the backup API not getting notified when essential K/V apps like @pm@ don't have new data for backup. The only option they offered is BackupMonitor which gets the information at least, but out of process on another thread.

 So we implemented a hacky workaround where BackupMonitor tells SnapshotCreator to extract backup data from an old snapshot.

 Unfortunately, it is possible that we do a backup run which includes @pm@, but encounters an error later, so the system cancels the entire backup which causes us not to have @pm@ data in a snapshots for re-use. Still, the system thinks we backed up @pm@ and doesn't give us its data.

grote committed:
Classify StreamResetException transient in WebDavBackend

grote committed:
Merge pull request #835 from grote/retry-backend

 Implement a auto re-trying wrapper backend

grote committed:
Halt AppBackupWorker progress if we get stopped or entered metered network

grote committed:
Treat SocketException as a transient exception in WebDavBackend

grote committed:
Improve app backup error notification

Add 'Try Again' button and show different text for metered network

grote committed:
Don't continue with APK backup if we move to metered network

grote committed:
Lazily get info in BackupRequester

so it only requests info when actually needed

grote committed:
Show dialog when starting backup on metered network

grote committed:
Abort app backups when we switched to metered network

We don't request new package chunks for backup and reject backup for all packages already requested.

grote committed:
Halt file backup when switched to metered network

grote committed:
Show proper error message when we couldn't find backups to restore

grote committed:
Also trust certificates the user added (self signed)

grote committed:
Log when RetryBackend will stop retrying and throw exception

grote committed:
Create codacy.yml

grote committed:
Create mobsf.yml

grote committed:
Replace head request with PROPFIND as not all WebDAV servers implement it

grote committed:
Log when we cancel backup notification so there can be no doubt we did it

Maybe helps with debugging stuck notifications

grote committed:
Bump quota to 3 GiB per app

grote committed:
Adapt metered network strings to feedback

grote committed:
Fix codacy complaints

grote committed:
Merge pull request #855 from seedvault-app/mobsf

Create mobsf.yml

grote committed:
Merge pull request #854 from seedvault-app/codacy

Create codacy.yml to see if their security scanner is helpful

weblate committed:
Import translations from Weblate

Co-authored-by: Andi Chandler <andi@gowling.com>
Co-authored-by: Anonymous <noreply@weblate.org>
Co-authored-by: Fjuro <fjuro@alius.cz>
Co-authored-by: Hasáky gum <caro.jf@protonmail.com>
Co-authored-by: Hosted Weblate <hosted@weblate.org>
Co-authored-by: Idcrafter <minefater1@gmail.com>
Co-authored-by: Igor Sorocean <sorocean.igor@gmail.com>
Co-authored-by: Jerguš Fonfer <caro.jf@protonmail.com>
Co-authored-by: Kristian Nordin <kristian.po@proton.me>
Co-authored-by: Languages add-on <noreply-addon-languages@weblate.org>
Co-authored-by: LucasMZ <git@lucasmz.dev>
Co-authored-by: Michael Bestas <mkbestas@gmail.com>
Co-authored-by: Milo Ivir <mail@milotype.de>
Co-authored-by: Oğuz Ersen <oguz@ersen.moe>
Co-authored-by: Renko <kull@tutanota.de>
Co-authored-by: Ricky Tigg <ricky.tigg@gmail.com>
Co-authored-by: Serhii Horichenko <serhii@horichenko.com>
Co-authored-by: Yaron Shahrabani <sh.yaron@gmail.com>
Co-authored-by: abc0922001 <abc0922001@hotmail.com>
Co-authored-by: gallegonovato <fran-carro@hotmail.es>
Co-authored-by: hhs bbsb <bamboo@ik.me>
Co-authored-by: hsfg6 <hsfg6@proton.me>
Co-authored-by: ikya91 <weblate.haq9g@simplelogin.com>
Co-authored-by: lucasmz-dev <git@lucasmz.dev>
Co-authored-by: nautilusx <translate@disroot.org>
Co-authored-by: nobody <info@localcdn.org>
Co-authored-by: pa cs <cs.pa@users.noreply.hosted.weblate.org>
Co-authored-by: rehork <cooky@e.email>
Co-authored-by: sysxda <coderz35@gmail.com>
Co-authored-by: Максим Горпиніч <mgorpinic2005@gmail.com>
Co-authored-by: தமிழ்நேரம் <anishprabu.t@gmail.com>
Co-authored-by: さしみ <kazasiuun@gmail.com>
Co-authored-by: 大王叫我来巡山 <hamburger2048@users.noreply.hosted.weblate.org>
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault/ca/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault/cs/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault/de/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault/el/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault/en_GB/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault/es/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault/fi/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault/fr/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault/he/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault/hr/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault/ja/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault/nl/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault/pl/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault/pt_BR/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault/ro/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault/ru/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault/sk/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault/sv/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault/ta/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault/tr/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault/uk/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault/zh_Hans/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault/zh_Hant/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_contactsbackup/en_GB/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_contactsbackup/hr/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_contactsbackup/sk/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_contactsbackup/ta/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/cs/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/de/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/el/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/en_GB/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/es/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/he/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/ja/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/pl/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/pt_BR/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/ro/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/sk/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/sv/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/ta/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/tr/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/uk/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/zh_Hans/
Translation: CalyxOS/Seedvault
Translation: CalyxOS/Seedvault contacts backup
Translation: CalyxOS/Seedvault storage backup

grote committed:
Merge pull request #845 from grote/no-data-no-snapshot

Tell system to reinitialize when it refuses to give us data we need

grote committed:
Upgrade actions/upload-artifact to v4

chirayudesai committed:
Merge pull request #865 from grote/ci-upload-fix

Upgrade actions/upload-artifact to v4

lucasmz-dev committed:
Bump version to 15-5.3

Check CHANGELOG for more details

lucasmz-dev committed:
Miscellaneous improvements to the README

- Uses [!WARNING] for better highlighting
- Adds a translation percentage next to the build status
- Adds a widget showing the status of all languages under Translating

grote committed:
Merge pull request #857 from grote/davx-webdav-improvements

Several assorted improvements

grote committed:
Also abort file backup while backup up a single huge file

previously switching to a metered network would have left the upload to continue until the file would be complete

grote committed:
Merge pull request #849 from grote/metered-network

Try everything to abort backups on metered network (when not allowed)

lucasmz-dev committed:
Reformat CHANGELOG to not use brackets

grote committed:
Merge pull request #871 from lucasmz-dev/readme

Miscellaneous improvements to the README

grote committed:
Merge pull request #870 from lucasmz-dev/next-version

Bump version to 15-5.3

mikeNG committed:
Merge pull request #668 from weblate/weblate-calyxos-seedvault

Translations update from Hosted Weblate

lucasmz-dev committed:
strings: Rename storage backup to file backup

lucasmz-dev committed:
strings: Fix a few strings translatable settings

grote committed:
Merge pull request #873 from lucasmz-dev/strings-rework

Fix a few issues with strings

grote committed:
Use different allow-lists for backup and restore

weblate committed:
Import translations from Weblate

Co-authored-by: Abduqadir Abliz <sahranbay@gmail.com>
Co-authored-by: Anonymous <noreply@weblate.org>
Co-authored-by: Fjuro <fjuro@users.noreply.hosted.weblate.org>
Co-authored-by: Hosted Weblate <hosted@weblate.org>
Co-authored-by: Igor Sorocean <sorocean.igor@gmail.com>
Co-authored-by: LucasMZ <git@lucasmz.dev>
Co-authored-by: Michael Bestas <mike+weblate@bestas.gr>
Co-authored-by: rehork <cooky@e.email>
Co-authored-by: Максим Горпиніч <maksimgorpinic2005a@gmail.com>
Co-authored-by: 大王叫我来巡山 <hamburger2048@users.noreply.hosted.weblate.org>
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault/cs/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault/el/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault/pl/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault/pt_BR/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault/ro/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault/ug/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault/uk/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault/zh_Hans/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_contactsbackup/pt_BR/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_contactsbackup/ug/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/ar/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/ca/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/cs/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/de/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/el/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/en_AU/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/en_GB/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/es/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/es_US/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/fr/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/gl/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/he/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/it/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/ja/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/ko/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/lt/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/ml/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/nb_NO/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/nl/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/pl/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/pt/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/pt_BR/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/ro/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/ru/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/sk/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/sv/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/ta/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/tr/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/ug/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/uk/
Translate-URL: https://hosted.weblate.org/projects/calyxos/seedvault_storage/zh_Hans/
Translation: CalyxOS/Seedvault
Translation: CalyxOS/Seedvault contacts backup
Translation: CalyxOS/Seedvault storage backup

grote committed:
Merge pull request #874 from grote/backend-allow-list-split

Use different allow-lists for backup and restore

chirayudesai committed:
Merge pull request #872 from weblate/weblate-calyxos-seedvault

Translations update from Hosted Weblate

chirayudesai committed:
Merge tag '15-5.3' of https://github.com/seedvault-app/seedvault

Seedvault 15-5.3

* Added support for user CA certificates
* Fixed issue where many Go server implementations of WebDAV did not work with the WebDAV client in Seedvault
* Bumped the app data quota from 1GB to 3GB
* Nextcloud app is no longer allowed for backup (Use built-in WebDAV Cloud support!)
* Improved handling of metered networks, if disallowed, the backup process will be aborted
* Fixed backup errors with USB when file and app backup are both on
* Fixed overdue backups not automatically starting when USB drives are plugged in
* The size of each app backup is now shown on the restore screen
* Fixed a common error (StaleDataException) causing backups to fail
* Fixed error message when no backups are available to restore
* Implemented a wrapper for the backend with a retrying mechanism, giving us less common errors
* Updated dependencies for 15 QPR1

Change-Id: Ic578a0f48df6422a45749e34d07f9cba733b874a

# -----BEGIN PGP SIGNATURE-----
#
# iQIzBAABCgAdFiEE8LiYseXDVwsHMtSKo5wwJqfeYAEFAme13QgACgkQo5wwJqfe
# YAHtFw//eCXG2DTC93f/y0Yqrm2QrVWCml7xZ0ZYcNO6fI0Jff3h5u3CX0wKItrk
# V5Qsj5o7EZQDzzQ1saOEHS/LbeFhIemMHvfAybAbB66WdW/PTr4KbHwgQIqwUWtO
# HxAI1OqGngNh1PtFTXIixqftQM7Sgx6xG9v1YdgAi7Pw7BN6v/7mFZL6fpaTHQOE
# +k/BYqxiEqKs6HnB/H8KRrryQtQz8Gyr8e6K3oMtDogkIIOk6y7JVIc/ejoQo3Yq
# 0wpaM0rhWYmklz4blw/JS4teNY3clG4OJn5AG33//reQCOFhHQfyr+WWX13VSf44
# rQt5VyzvR3PqOqwZjAhtyUgFLPf+bbDGlUsmOMB+4Nnxb1caVGWG5N7IcfNgC+L3
# VksSHrLmYKV39dzl9KwDJVNoOJMBgoQDtG7U34x5O+NMDw2RHKhpmcg9Nnfqqn7G
# t8cVGdUVD6n+HCsGakRrYzOjISgDT7psSdIKbaerSfkjkqE1poe5UX7ynZZgSXdo
# 3SZMitdg9o+BbstkLdx8uPNgb6Yl2qiRAMa6KO7ukhTflPko+Q80tUkRxuIVnWhe
# lA5iW/h8Vp1I1aVDKoy/06fS8yniGGzzlhnlKHjaDRI1QZgILAkzAKyUn4d5DrXk
# G7mWzC/8TYmVp/N4FczdzvNfgvD+nskzvFsLmcPIfLVddjYJnEc=
# =MfPW
# -----END PGP SIGNATURE-----
# gpg: Signature made Wed 19 Feb 2025 07:00:48 PM IST
# gpg:                using RSA key F0B898B1E5C3570B0732D48AA39C3026A7DE6001
# gpg: Good signature from "Chirayu Desai <chirayudesai1@gmail.com>" [ultimate]
# gpg:                 aka "Chirayu Desai <chirayu@calyxinstitute.org>" [ultimate]
# gpg:                 aka "Chirayu Desai <chirayu@cdesai.in>" [ultimate]

grote commented:
This seems to include  [Update libs for Android 15 QPR1](https://github.com/seedvault-app/seedvault/pull/878/commits/bc575ac1d064498816380986e99a1675b304858c). What did you do? Merge the entirety of Android 15 branch into 14?

chirayudesai commented:
> This seems to include [Update libs for Android 15 QPR1](https://github.com/seedvault-app/seedvault/pull/878/commits/bc575ac1d064498816380986e99a1675b304858c).

It includes the commit, but I undid the changes in the merge commit, so you shouldn't see them in the full diff.

> What did you do? Merge the entirety of Android 15 branch into 14?

I merged the signed tag, see [Merge tag '15-5.3'](https://github.com/seedvault-app/seedvault/pull/878/commits/1ed56524c77df406a9313f24608c3efabaec35b3)

grote commented:
Something's wrong though: `e: file:///home/runner/work/seedvault/seedvault/app/src/main/java/com/stevesoltys/seedvault/UsbMonitorService.kt:61:5 'onTimeout' overrides nothing`

chirayudesai committed:
fixup! Use UsbMonitorService in foreground to wait for USB root, for Android 14

* Fixes:
  `e: file:///home/runner/work/seedvault/seedvault/app/src/main/java/com/stevesoltys/seedvault/UsbMonitorService.kt:61:5 'onTimeout' overrides nothing`
* This method was only added in 15, per
  https://developer.android.com/reference/android/app/Service#onTimeout(int,%20int)

Change-Id: I4cbd2d06372f24ab8d86a29dec23b28dd2fb5ed8

chirayudesai commented:
> Something's wrong though: `e: file:///home/runner/work/seedvault/seedvault/app/src/main/java/com/stevesoltys/seedvault/UsbMonitorService.kt:61:5 'onTimeout' overrides nothing`

That's because this was only added in 15, per https://developer.android.com/reference/android/app/Service#onTimeout(int,%20int)

Removing that new method makes it build. Not sure if there's a better way to fix it, but it doesn't seem too bad.

grote commented:
> Removing that new method makes it build. Not sure if there's a better way to fix it, but it doesn't seem too bad.

Sounds good. 

Did anybody test this with an aosp build on 14? or do we yolo merge it?

chirayudesai commented:
> Did anybody test this with an aosp build on 14?

I'm doing that now, it built fine in AOSP even without this FTR.

Will merge after.

grote reviewed (approved):
fine with me if a smoke test on A14 passes

lucasmz-dev commented:
i'll go and test this later today then, regular user test

lucasmz-dev commented:
I went and restored my A15 backup on A14. That seemed to have worked, my wallpaper didn't get backed up but it seems the rest is working as expected

Now starting another backup from that version

lucasmz-dev commented:
Backup and restore using local storage seems to work (pulled/pushed files with ADB)
Now trying USB

lucasmz-dev commented:
Seems to work fine :)

grote commented:
Thanks for testing!
